src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -72,6 +72,10 @@
     public static final String IDP_METADATA_URL = "idp.metadata_url";
     public static final String IDP_METADATA_FILE = "idp.metadata_file";
     public static final String IDP_METADATA_CONTENT = "idp.metadata_content";
+    public static final String OPENDISTRO_AUTHTOKEN_PATH = "/_opendistro/_security/api/authtoken";
```

Comment:
Can you get this from `routes` of a corresponding class?

Reply:
`routes()` in Action classes return list and not a specific string, it returns a list of routes.

Reply:
each element in the list has method `getPath()` that returns `String`. Note that I do not suggest to use `route()` method, but use `routes` directly.

Reply:
Sorry but I am not sure what is the benefit of it? We need two paths one for `_opendistro` and the other one for `_plugins`, if we use routes, then we need to get all the paths from the routes, find unique ones(by making a set) and use them for string comparing.Is it really necessary when we only have two possible routes?

Reply:
The benefit is to avoid string duplication in multiple places. It is easy to change the string in one place and forget to change it in another place.

Reply:
I got this, but given the time constraint, I believe we can do it later as an improvement.

Reply:
OK, at minimum change to compiled pattern.

Reply:
In this code, I need exact matching, not regx matching. I have 4 different strings, so not sure if I can use regx or compiled pattern here, please let me know if you are looking for something else.

Reply:
Pattern and regex allow to match exact strings as well. Use 2 compiled patterns to match 2 sets of strings.

Reply:
I did what you suggested, however in my opinion the code is not readable anymore.

Reply:
What exactly makes code not readable?